mephisto/core/task_launcher.py

```diff
@@ -10,20 +10,40 @@
 # interface could be like an iterator. This class will launch tasks
 # as if the loader is an iterator.
 
-from mephisto.data_model.assignment import Assignment, Unit, InitializationData
+from mephisto.data_model.assignment import (
+    Assignment,
+    Unit,
+    InitializationData,
+    AssignmentState,
+)
 
 from typing import Dict, Optional, List, Any, TYPE_CHECKING
 
 import os
+import time
+from datetime import datetime
 
 if TYPE_CHECKING:
     from mephisto.data_model.task import TaskRun
     from mephisto.data_model.database import MephistoDB
 
+import threading
 from mephisto.core.logger_core import get_logger
 
 logger = get_logger(name=__name__, verbose=True, level="info")
 
+UNIT_GENERATOR_WAIT_SECONDS = 10
+
+
+class LimitedDict(dict):
+    def __init__(self, limit):
+        self.limit = limit
+        super().__init__()
+
+    def __setitem__(self, key, value):
+        super().__setitem__(key, value)
+        assert self.limit, len(self.keys())
```

Comment:
I'm not sure what you're trying to do with this assert - the `assert` syntax is `assert (conditional), (message for if conditional fails)`